tldextract/remote.py

```diff
@@ -29,6 +25,23 @@ def lenient_netloc(url: str) -> str:
     )
 
 
+def lenient_netloc(url: str) -> str:
+    """Extract the netloc of a URL-like string, similar to the netloc attribute
+    returned by urllib.parse.{urlparse,urlsplit}, but extract more leniently,
+    without raising errors."""
+
+    double_slashes_start = url.find("//")
+    if double_slashes_start == 0:
+        return extract_netloc(url[2:])
+    if (
+        double_slashes_start < 2
```

Comment:
What conditions are these 3 lines testing? If I delete either or both of the 2nd and 3rd lines, tests still pass. Can you add tests?

Reply:
```python
>       assert_extract(
            "//mail.google.com/mail", ("mail.google.com", "mail", "google", "com")
        )
```

The above test fails on Python 3.10 after commenting out these 2 lines. Are you getting the same results?

```python
    # if double_slashes_start == 0:
    #     return url[2:]
```

Reply:
I think you meant

```python
        # or not url[double_slashes_start - 1] == ":"
        # or set(url[: double_slashes_start - 1]) - scheme_chars_set
```

Reply:
I've added 2 tests for the above 2 conditions.

`tldextract.extract("a#//example.com")` output is the same as the regex version.